docs/reference-architectures/containers/aks-mission-critical/mission-critical-data-platform.md

```diff
@@ -157,6 +157,11 @@ Azure Service Bus premium tier is the recommended solution for high-value messag
   - If an acknowledgment isn't received by the broker in the allotted time period, or the handler explicitly abandons the message, the exclusive lock is released. The message is then available for other consumers to process the message.
   - If a message is not successfully processed a configurable number of times, or the handler forwards the message to the dead-letter queue.
 - Because messages can potentially be processed more than one time, message handlers should be made idempotent.
+  - "A ... method is considered `idempotent` if the intended effect on the server of multiple identical requests with that method is the same as the effect for a single such request." ([RFC 7231](https://tools.ietf.org/html/rfc7231#section-4))
+  - One common technique of making message handling idempotent is to check some persistent store, like a database, if the message has already been processed, and if so, not running the logic to process it again. 
+  - However, if the processing of the message includes database operations, specifically the insertion of new records with database-generated identifiers, as well as the emitting of new messages to the broker containing those identifiers, because there aren't distributed transactions which encompass both the database and the message broker, there can be a number of complications that can occur if the process running the code happens to fail, for example:
+    - If the code emitting the messages runs before the database transaction is committed, which is how many developers work using the Unit of Work pattern, those messages can "escape" if the failure occurs between calling the broker and asking that the database transaction be committed. As the transaction rolls back, those database-generated IDs are also undone, leaving them available to other code which may be running at the same time. This can cause recipients of the "escaped" messages to processes the wrong database entries, hurting the overall consistency and correctness of your system.
```

Comment:
This describes the complications, but does't provide guidance. What's the recommendation here?

Reply:
Added recommendations.